### ISSUE TITLE
False positive type mismatch: #[cfg]-gated panic in tail position

### ISSUE THREAD

**Issue by benediktsatalia** [rust-lang/rust-analyzer#22598]
**rust-analyzer version**: `0.3.2937-standalone`

**rustc version**: `rustc 1.94.0 (4a4ef493e 2026-03-02)`

**editor or extension**: VSCode

**code snippet to reproduce**:
```rust
fn returns_i64() -> i64 {
    #[cfg(feature = "some_feature")]
    {
        5
    }
    #[cfg(not(feature = "some_feature"))]
    panic!()
}
```

The compiler is fine with this code, but rust-analyzer fails if run with the `some_feature` feature activated (I am using `"rust-analyzer.check.features": "all"` but I think it also happens if `some_feature` is part of the default features) with the following error pointing to the `5`:

```
expected (), found i32 (E0308)
```

Interestingly, using a block around `panic!()` or adding a semicolon after it solves the problem.
